src/rust/inetstack/protocols/tcp/tests/simulator.rs

```diff
@@ -876,21 +882,36 @@ impl Simulation {
         let ipv4_hdr: Ipv4Header = self.build_ipv4_header(IpProtocol::UDP);
         let udp_hdr: UdpHeader = self.build_udp_header(&udp_packet);
         let data: DemiBuffer = Self::cook_buffer(udp_packet.len as usize, None);
-        let mut datagram: UdpDatagram = UdpDatagram::new(ipv4_hdr, udp_hdr, data, false).unwrap();
+        // This is an incoming packet, so the source is the remote address and the destination is the local address.
+        let mut datagram: UdpDatagram = UdpDatagram::new(
+            &self.remote_sockaddr.ip(),
+            &self.local_sockaddr.ip(),
+            udp_hdr,
+            data,
+            false,
+        )
+        .unwrap();
 
         let mut pkt: DemiBuffer = datagram.take_body().unwrap();
-
-        // Actually prepend L2 header.
+        let payload_bytes: usize = pkt.len();
```

Comment:
This should be renamed to num_payload_bytes or something similar. For a sec I thought these are the actual bytes. :)

Reply:
Ah yes, definitely!